clients/client-java/src/main/java/com/datastrato/gravitino/client/GravitinoFilesetCatalog.java

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright 2024 Datastrato Pvt Ltd.
+ * This software is licensed under the Apache License version 2.
+ */
+package com.datastrato.gravitino.client;
+
+import com.datastrato.gravitino.NameIdentifier;
+import com.datastrato.gravitino.Namespace;
+import com.datastrato.gravitino.dto.AuditDTO;
+import com.datastrato.gravitino.dto.requests.FilesetCreateRequest;
+import com.datastrato.gravitino.dto.requests.FilesetUpdateRequest;
+import com.datastrato.gravitino.dto.requests.FilesetUpdatesRequest;
+import com.datastrato.gravitino.dto.responses.DropResponse;
+import com.datastrato.gravitino.dto.responses.EntityListResponse;
+import com.datastrato.gravitino.dto.responses.FilesetResponse;
+import com.datastrato.gravitino.exceptions.FilesetAlreadyExistsException;
+import com.datastrato.gravitino.exceptions.NoSuchFilesetException;
+import com.datastrato.gravitino.exceptions.NoSuchSchemaException;
+import com.datastrato.gravitino.file.Fileset;
+import com.datastrato.gravitino.file.FilesetCatalog;
+import com.datastrato.gravitino.file.FilesetChange;
+import com.datastrato.gravitino.rest.RESTUtils;
+import com.google.common.annotations.VisibleForTesting;
+import com.google.common.base.Preconditions;
+import java.util.Arrays;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+import org.apache.commons.lang3.StringUtils;
+
+/**
+ * Fileset catalog is a catalog implementation that supports fileset like metadata operations, for
+ * example, schemas and filesets list, creation, update and deletion. A Fileset catalog is under the
+ * metalake.
+ */
+public class GravitinoFilesetCatalog extends GravitinoCatalog implements FilesetCatalog {
```

Comment:
Adding "Gravitino" to the class name seems a bit weird, can we figure out a better name?

Reply:
@jerryshao The name `FilesetCatalog` is the same with `com.datastrato.gravitino.file.FilesetCatalog`. If we do not rename `FILE` to `FILESET` in the enum, we can call it `FileCatalog`. Do you have any advice?